File: ISA_BACKEND/BloodBank/src/main/java/com/example/bloodbank/Model/DonorSurvey.java
package com.example.bloodbank.Model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "DonorSurvey")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class DonorSurvey {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private long id;
    @OneToOne
    @JoinColumn(name = "blood_donor_id")
    private BloodDonor BloodDonor;
    private Boolean isAvailable;
}

File: ISA_FRONTEND/src/app/model/donorSurvey.ts
import { BloodDonor } from "./bloodDonor";

export class DonorSurvey {
    id: string = '';
    bloodDonor: BloodDonor = new BloodDonor;
    isAvailable: Boolean = true;

    public constructor(obj?: any) {
        if (obj) {
            this.id = obj.id;
            this.bloodDonor = obj.bloodDonor;
            this.isAvailable = obj.isAvailable;
        }
    }
}